spec/controllers/events_controller_spec.rb

```diff
@@ -362,6 +378,120 @@
     end
   end
 
+  describe 'GET League#ranking' do
+    it 'should calculate an empty ranking when no participant has joined the event' do
+      event = League.create! valid_league_attributes
+      get :ranking, params: { id: event.to_param }, session: valid_session
+      ranking_entries = controller.instance_variable_get(:@ranking_entries)
+      expect(ranking_entries).to be_empty
+    end
+    describe 'when teams have joined the event' do
+      describe 'when no match has been played yet' do
+        it 'should calculate a lexicographically sorted zero-filled ranking' do
+          event = League.create! valid_league_attributes
+          team1 = FactoryBot.create(:team)
+          team2 = FactoryBot.create(:team)
+          event.add_team(team1)
+          event.add_team(team2)
+          event.generate_schedule
```

Comment:
Why do you need to generate a schedule in `when no match has been played yet` but not in the following test cases in `when at least two matches have been played already`?